### PR TITLE
Add role cleanup

### PR DESCRIPTION
**Pull request by sobolk**
## Problem

Partially solves https://github.com/aws-amplify/amplify-backend/issues/918 .
I.e. adds orphan role removal to cleanup script.

**Issue number, if available:**

## Changes

Add role removal to cleanup script.

**Corresponding docs PR, if applicable:**

## Validation

https://github.com/aws-amplify/amplify-backend/actions/workflows/e2e_resource_cleanup.yml?query=branch%3Aadd-role-cleanup

## Checklist

- [ ] If this PR includes a functional change to the runtime behavior of the code, I have added or updated automated test coverage for this change.
- [ ] If this PR requires a change to the [Project Architecture README](../PROJECT_ARCHITECTURE.md), I have included that update in this PR.
- [ ] If this PR requires a docs update, I have linked to that docs PR above.
- [ ] If this PR modifies E2E tests, makes changes to resource provisioning, or makes SDK calls, I have run the PR checks with the `run-e2e` label set.

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

**Review by edwardfoyle — APPROVED** [review]

File: scripts/cleanup_e2e_resources.ts [hunk 1]
```diff
@@ -342,3 +360,77 @@ for (const staleBranch of allStaleBranches) {
...
+      new ListRolesCommand({
+        Marker: nextToken,
+      })
+    );
+    nextToken = listRolesCommandOutput.Marker;
+    if (listRolesCommandOutput.Roles) {
+      listRolesCommandOutput.Roles.filter(
+        (role: Role) =>
+          (role.RoleName?.startsWith(TEST_AMPLIFY_RESOURCE_PREFIX) ||
+            role.RoleName?.startsWith(TEST_CDK_RESOURCE_PREFIX)) &&
```
Comment (rtpascual) [inline scripts/cleanup_e2e_resources.ts:378]: Are roles the only test resources that can have `TEST_CDK_RESOURCE_PREFIX`? Wondering if this should be extended to other resources as well.

Comment (sobolk) [reply scripts/cleanup_e2e_resources.ts:378]: yes, these come from auth stacks only so far.

Comment (sobolk) [reply scripts/cleanup_e2e_resources.ts:378]: We don't orphan other resources from cdk tests so far, so we'll expand this later as needed.

**Review by rtpascual — APPROVED** [review]

File: scripts/cleanup_e2e_resources.ts [hunk 2]
```diff
@@ -342,3 +360,77 @@ for (const staleBranch of allStaleBranches) {
...
+      if (
+        policy.PolicyArn &&
+        !policy.PolicyArn.startsWith('arn:aws:iam::aws:policy')
+      ) {
+        // delete non AWS managed policies as well
+        await iamClient.send(
+          new DeletePolicyCommand({ PolicyArn: policy.PolicyArn })
+        );
+      }
+    }
```
[the remark this answers is not in the document]
Comment (sobolk) [reply scripts/cleanup_e2e_resources.ts]: Some facts:
1. these items here are from listing of attached roles to `amplif-*` roles, which are most likely test resources
2. on the other hand after running health_checks on main there doesn't seem to be trace of orhpaned policies at all.

Therefore removing this for now.

**Review by rtpascual — APPROVED** [review]

**Review by Amplifiyer — APPROVED** [review]

**Pull request merged**